core/src/test/java/org/apache/brooklyn/core/entity/lifecycle/ServiceStateLogicTest.java

```diff
@@ -281,6 +291,35 @@ public void testQuorumWithStringStates() {
         EntityAsserts.assertAttributeEqualsContinually(cluster, Attributes.SERVICE_STATE_ACTUAL, Lifecycle.RUNNING);
     }
 
+    public static class CountingComputeServiceState extends ComputeServiceState {
+        AtomicInteger cntCalled = new AtomicInteger();
+        AtomicInteger cntCalledWithNull = new AtomicInteger();
+
+        public CountingComputeServiceState() {}
+
+        @Override
+        public void onEvent(SensorEvent<Object> event) {
+            cntCalled.incrementAndGet();
+            if (event == null) {
+                cntCalledWithNull.incrementAndGet();
+            }
+            super.onEvent(event);
+        }
+    }
+
+    @Test
+    public void testServiceStateNotCalledExplicitly() throws Exception {
```

Comment:
Wasn't able to come up with a deterministic test to observe the failure case. Instead I'm asserting that the cause is not happening any more.